Kernel/Net/VirtIO/VirtIONetworkAdapter.cpp

```diff
@@ -0,0 +1,284 @@
+static constexpr u64 VIRTIO_NET_F_GUEST_UFO = (1ull << 10);          // Driver can receive UFO.
+static constexpr u64 VIRTIO_NET_F_HOST_TSO4 = (1ull << 11);          // Device can receive TSOv4.
+static constexpr u64 VIRTIO_NET_F_HOST_TSO6 = (1ull << 12);          // Device can receive TSOv6.
+static constexpr u64 VIRTIO_NET_F_HOST_ECN = (1ull << 13);           // Device can receive TSO with ECN.
+static constexpr u64 VIRTIO_NET_F_HOST_UFO = (1ull << 14);           // Device can receive UFO.
+static constexpr u64 VIRTIO_NET_F_MRG_RXBUF = (1ull << 15);          // Driver can merge receive buffers.
+static constexpr u64 VIRTIO_NET_F_STATUS = (1ull << 16);             // Configuration status field is available.
+static constexpr u64 VIRTIO_NET_F_CTRL_VQ = (1ull << 17);            // Control channel is available.
+static constexpr u64 VIRTIO_NET_F_CTRL_RX = (1ull << 18);            // Control channel RX mode support.
+static constexpr u64 VIRTIO_NET_F_CTRL_VLAN = (1ull << 19);          // Control channel VLAN filtering.
+static constexpr u64 VIRTIO_NET_F_GUEST_ANNOUNCE = (1ull << 21);     // Driver can send gratuitous packets.
+static constexpr u64 VIRTIO_NET_F_MQ = (1ull << 22);                 // Device supports multiqueue with automatic receive steering.
+static constexpr u64 VIRTIO_NET_F_CTRL_MAC_ADDR = (1ull << 23);      // Set MAC address through control channel.
+static constexpr u64 VIRTIO_NET_F_HOST_USO = (1ull << 56);           // Device can receive USO packets.
+static constexpr u64 VIRTIO_NET_F_HASH_REPORT = (1ull << 57);        // Device can report per-packet hash value and a type of calculated hash.
+static constexpr u64 VIRTIO_NET_F_GUEST_HDRLEN = (1ull << 59);       // Driver can provide the exact hdr_len value.
+static constexpr u64 VIRTIO_NET_F_RSS = (1ull << 60);                // Device supports RSS with Toeplitz hash calculation
+static constexpr u64 VIRTIO_NET_F_RSC_EXT = (1ull << 21);            // Device can process duplicated ACKs and report number of coalesced segments and duplicated ACKs.
+static constexpr u64 VIRTIO_NET_F_STANDBY = (1ull << 63);            // Device may act as a standby for a primary device with the same MAC address.
+static constexpr u64 VIRTIO_NET_F_SPEED_DUPLEX = (1ull << 63);       // Device reports speed and duplex.
+
+static constexpr u16 VIRTIO_NET_S_LINK_UP = 1;
+static constexpr u16 VIRTIO_NET_S_ANNOUNCE = 2;
+
+static constexpr u8 VIRTIO_NET_HDR_F_NEEDS_CSUM = 1;
+static constexpr u8 VIRTIO_NET_HDR_F_DATA_VALID = 1;
+static constexpr u8 VIRTIO_NET_HDR_F_RSC_INFO = 1;
+static constexpr u8 VIRTIO_NET_HDR_GSO_NONE = 0;
+static constexpr u8 VIRTIO_NET_HDR_GSO_TCPV4 = 1;
+static constexpr u8 VIRTIO_NET_HDR_GSO_UDP = 3;
+static constexpr u8 VIRTIO_NET_HDR_GSO_TCPV6 = 4;
+static constexpr u8 VIRTIO_NET_HDR_GSO_UDP_L4 = 5;
+static constexpr u8 VIRTIO_NET_HDR_GSO_ECN = 0x80;
+
+struct [[gnu::packed]] virtio_net_config {
```

Comment:
These two structs lend their names directly from the spec, but our coding style dictates a camel case name for structs and classes. I think that as long as the name is largely the same, it's still clear what part of the spec it refers to.